eng/native/configuretools.cmake

```diff
@@ -40,7 +40,7 @@ if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_ARCH_WASM)
   endfunction()
 
   locate_toolchain_exec(ar CMAKE_AR)
-  locate_toolchain_exec(link CMAKE_LINKER)
+  locate_toolchain_exec(ld CMAKE_LINKER)
```

Comment:
`link(2)` is something else (file alias) on Unix. `CMAKE_LINKER` has no effect on GNU or LLVM toolchain.

context: I was trying to switch to GNU linker on SmartOS at the beginning, but now the native linker works well, so we do not need to force the tool on Illumos. When we will switch to "build dotnet with LLVM" on Illumos, at that point we can detect which linker is selected and make decision on setting linker options.